android-example/src/java_glue.rs.in

```diff
@@ -4,5 +4,6 @@ use Session;
 foreigner_class!(class Session {
     self_type Session;
     constructor Session::new() -> Session;
-    method Session::test(&self, val: i32) -> i32;
+    method Session::add_and1(&self, val: i32) -> i32;
+    static_method Session::greet(to: &String) -> String;
```

Comment:
I can accept a `&str`, but I need to return an owned String to have enough lifetime (I think).

Reply:
I request only input parameter change, because of it is really speedup, plus it is more Rust'y
with `&str` as input you get:
```rust
//pseudo code
auto x: *const c_char = GetStringUTFChars(jstring);
auto x: &str = magic(x);
rust_function(x);
ReleaseStringUTFChars(x);
```

one allocation via `GetStringUTFChars`,
with `&String` you get to two  allocations, one `GetStringUTFChars` another
via `String::new`.

Return `-> String` is right decision and I have no complains about it.

Reply:
Thanks! I'm still working on learning rust and didn't know that.